### PR TITLE
Bump LLVM to 469c58944d4f0f990ca8ee88ebcc37624801a36c.

### PR DESCRIPTION
**Pull request by gysit**
- changes around func op
- updated to new interface registration

### CODE REVIEW

File: lib/Dialect/LinalgExt/IR/TiledOpInterface.cpp [hunk 1]
```diff
@@ -305,9 +306,11 @@ void IREE::LinalgExt::registerTiledOpInterfaceExternalModels(
     DialectRegistry &registry) {
   LLVM_DEBUG(
       { llvm::dbgs() << "Adding external models of tiled op interface\n"; });
-  registry
-      .addOpInterface<tensor::ExtractSliceOp, ExtractSliceTiledOpInterface>();
-  registry.addOpInterface<tensor::InsertSliceOp, InsertSliceTiledOpInterface>();
+  registry.addExtension(+[](MLIRContext *ctx,
```
Comment (nicolasvasilache) [inline lib/Dialect/LinalgExt/IR/TiledOpInterface.cpp:309]: I see `addExtension(+[](`, is the `+` a typo?

File: lib/Dialect/LinalgExt/Transforms/LinalgExtBufferization.cpp [hunk 2]
```diff
@@ -341,9 +342,11 @@ struct ParallelInsertSliceOpInterface
 
 void mlir::iree_compiler::IREE::LinalgExt::
     registerBufferizableOpInterfaceExternalModels(DialectRegistry &registry) {
-  registry.addOpInterface<InParallelOp, InParallelOpInterface>();
-  registry
-      .addOpInterface<PerformConcurrentlyOp, PerformConcurrentlyOpInterface>();
-  registry
-      .addOpInterface<ParallelInsertSliceOp, ParallelInsertSliceOpInterface>();
+  registry.addExtension(+[](MLIRContext *ctx, IREELinalgExtDialect *dialect) {
```
Comment (nicolasvasilache) [inline lib/Dialect/LinalgExt/Transforms/LinalgExtBufferization.cpp:345]: Same Q here, I do not know this `(+[]` form ?

Comment (nicolasvasilache) [reply lib/Dialect/LinalgExt/Transforms/LinalgExtBufferization.cpp:345]: Ah nm, found it: https://stackoverflow.com/questions/18889028/a-positive-lambda-what-sorcery-is-this

**Review by nicolasvasilache — APPROVED** [review]

File: lib/Dialect/LinalgExt/Transforms/LinalgExtBufferization.cpp (hunk 2 above)
Comment (gysit) [reply lib/Dialect/LinalgExt/Transforms/LinalgExtBufferization.cpp:345]: also saw this for the first time yesterday...

File: lib/Dialect/LinalgExt/IR/TiledOpInterface.cpp (hunk 1 above)
Comment (chelini) [reply lib/Dialect/LinalgExt/IR/TiledOpInterface.cpp:309]: No it is not: https://stackoverflow.com/questions/18889028/a-positive-lambda-what-sorcery-is-this

**Pull request merged**